xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp

```diff
@@ -616,6 +692,14 @@ void CGUIDialogPVRChannelManager::Update()
     SET_CONTROL_LABELS(SPIN_EPGSOURCE_SELECTION, 0, &labels);
   }
 
+  m_clientsWithSettingsList = g_PVRClients->GetClientsSupportingChannelSettings(m_bIsRadio);
+  if (!m_clientsWithSettingsList.empty())
+    m_bAllowNewChannel = true;
+
+  CGUIButtonControl *btnShowNewChannel = (CGUIButtonControl*) GetControl(BUTTON_NEW_CHANNEL);
+  if (btnShowNewChannel)
+    btnShowNewChannel->SetVisible(m_bAllowNewChannel);
```

Comment:
As I said on IRC you should use the macro's SET_CONTROL_HIDDEN() and  SET_CONTROL_VISIBLE().

```
if (m_bAllowNewChannel)
  SET_CONTROL_VISIBLE(BUTTON_NEW_CHANNEL);
else
  SET_CONTROL_HIDDEN(BUTTON_NEW_CHANNEL);
```